x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_sync_rules_table_saved_state.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { URL_PARAM_KEY } from '../../../../../common/hooks/use_url_state';
+import { useUpdateUrlParam } from '../../../../../common/utils/global_query_string';
+import { RULES_TABLE_STATE_STORAGE_KEY } from '../constants';
+import { useRulesTableContext } from './rules_table_context';
+import type { RulesTableSavedState } from './rules_table_saved_state';
+import {
+  DEFAULT_PAGE,
+  DEFAULT_RULES_PER_PAGE,
+  DEFAULT_SORTING_OPTIONS,
+} from './rules_table_defaults';
+
+export function useSyncRulesTableSavedState(): void {
+  const { state } = useRulesTableContext();
+  const {
+    services: { sessionStorage },
+  } = useKibana();
+  const updateUrlParam = useUpdateUrlParam<RulesTableSavedState>(URL_PARAM_KEY.rulesTable);
+
+  useEffect(() => {
+    const savedState: RulesTableSavedState = {};
+
+    if (state.filterOptions.filter.length > 0) {
+      savedState.searchTerm = state.filterOptions.filter;
+    }
+
+    if (state.filterOptions.showElasticRules || state.filterOptions.showCustomRules) {
+      savedState.showCustomRules = state.filterOptions.showCustomRules;
+    }
+
+    if (state.filterOptions.tags.length > 0) {
+      savedState.tags = state.filterOptions.tags;
+    }
+
+    if (state.sortingOptions.field !== DEFAULT_SORTING_OPTIONS.field) {
+      savedState.sort = { field: state.sortingOptions.field };
+    }
+
+    if (state.sortingOptions.order !== DEFAULT_SORTING_OPTIONS.order) {
+      savedState.sort = { ...savedState.sort, order: state.sortingOptions.order };
+    }
+
+    if (state.pagination.page !== DEFAULT_PAGE) {
+      savedState.page = state.pagination.page;
+    }
```

Comment:
Do we really want to persist the `page`?

I mean, from a UX point of view, after a reload, will users expect/notice the page was persisted, or will they expect it was reset to the first one?

Or, if I navigate to Rules from another page, after the rules table is loaded, does it makes sense to find myself on a page that is not the first one? I don't think it is obvious.

The rest of the parameters make a lot of sense, they are a kind of table "settings", but I don't feel the same way about the `page` parameter, even more considering that the selected rules aren't persisted either.

Just to have a think.

Reply:
Great point @semd, we'll think about that 👍

Reply:
@semd you have a valid point though it depends on the usage scenario. One of the problems state persistence solve is returning back from the rule details page. Expectation is the user sees the same page. Another scenario is link sharing where some user opening a link should see the specific page.

Different scenarios like returning from another page inside or outside the security plugin may cause some confusion but it's mitigated by using the session storage.

There is also a suggestion to give users a simple way to clear the table state for example a button next to the filter.

So we need to analyse that scenarios and decide what fits better.

Reply:
Okay, link sharing could still work with the specific page without persisting the value to the storage. But if you think it makes sense to have it, let's keep it 👍

Reply:
@semd we decided to omit the page number from the session storage after discussion with the team.